### PR TITLE
Remove FastAPIError class.

### PR DESCRIPTION
**Pull request by blkst8**

### CODE REVIEW

**Comment by SirTelemak** [comment]
Why?

File: fastapi/utils.py [hunk 1]
```diff
@@ -103,7 +103,7 @@ def create_response_field(
         else:  # pragma: nocover
             return response_field(schema=field_info)
     except RuntimeError:
-        raise fastapi.exceptions.FastAPIError(
+        raise RuntimeError(
             f"Invalid args for response field! Hint: check that {type_} is a valid pydantic field type"
         )
```
Comment (VaultVulp) [inline fastapi/utils.py:108]: Why should we stop with the `RuntimeError`? Better use `raise Exception`, for `RuntimeError` is a descendant of an `Exception` class.

Comment (MashhadiNima) [reply fastapi/utils.py:108]: `Exception` is general. Using `RuntimeError` when the exception is `RuntimeError` makes the code more intelligible.

**Comment by MashhadiNima** [comment]
> Why?

Because the class ‍‍‍"FastAPIError" inherits "RuntimeError" and doesn't have anything additional.
In my opinion, it would be better to use "RuntimeError" directly.

**Comment by SirTelemak** [comment]
> > Why?
> 
> Because the class ‍‍‍"FastAPIError" inherits "RuntimeError" and doesn't have anything additional.
> In my opinion, it would be better to use "RuntimeError" directly.

FastAPIError should'n add anything additional, it just helps you to find the source of mistake. RuntimeError is built-in and can be evoked by several other libraries and it will be hard to find exception source. But `FastAPIError` is specific for FastAPI. 
For example, you now your code can raise FastAPIError and what to mitigate this, you do something like this
```python
try:
    func_that_may_rise_error()
except FastAPIError:
    log.info('it\'s okay, we can handle this')
    handle_error()
except RuntimeError:
    log.exception('something weird happened, take a look')
    raise
```

File: fastapi/utils.py (hunk 1 above)
Comment (jansedlon) [reply fastapi/utils.py:108]: So does `FastAPIError`

**Comment by MashhadiNima** [comment]
Thanks for your explanation. 
My idea isn't good. sorry to all. I will close this PL.

**Comment by tiangolo** [comment]
Thanks for the help here @SirTelemak ! :cake: 

And thanks @MashhadiNima for the interest. :nerd_face:

**Pull request closed without merging**